### ISSUE TITLE
Dependabot ignore minor and major does not work for Dockerfile

### ISSUE THREAD

**Issue by gitflo1** [dependabot/dependabot-core#4085]
**Package ecosystem**
`docker`
**Manifest location and content prior to update**
`/nginx-1.20/Dockerfile`
**dependabot.yml content**
```
version: 2
updates:
  - package-ecosystem: "docker" # See documentation for possible values
    directory: "/nginx-1.20/" # Location of package manifests
    schedule:
      interval: "daily"
    ignore:
      # ignore all GitHub linguist patch updates
      - dependency-name: "*"
        update-types: ["version-update:semver-major", "version-update:semver-minor"]
```
**Updated dependency**
`nginx:1.20-alpine` (desired target dependency version is `nginx:1.20.1-alpine`)
**What you expected to see, versus what you actually saw**
- Expected behavior: Dependabot opens PR for bumping nginx:1.20-alpine version to 1.20.1-alpine
- Actual behavior: No PR is opened up

**Images of the diff or a link to the PR, issue or logs**
Dependabot logs:
```
service=registry.docker.io&scope=repository%3Alibrary%2Fnginx%3Apull
  proxy | 2021/07/26 07:53:07 [034] HEAD https://registry.hub.docker.com:443/v2/library/nginx/manifests/1.20-alpine
  proxy | 2021/07/26 07:53:07 [034] 200 https://registry.hub.docker.com:443/v2/library/nginx/manifests/1.20-alpine
  proxy | 2021/07/26 07:53:07 [036] HEAD https://registry.hub.docker.com:443/v2/library/nginx/manifests/1.20.1-alpine
  proxy | 2021/07/26 07:53:07 [036] 401 https://registry.hub.docker.com:443/v2/library/nginx/manifests/1.20.1-alpine
  proxy | 2021/07/26 07:53:07 [038] GET https://auth.docker.io:443/token?service=registry.docker.io&scope=repository%3Alibrary%2Fnginx%3Apull
  proxy | 2021/07/26 07:53:07 [038] 200 https://auth.docker.io:443/token?service=registry.docker.io&scope=repository%3Alibrary%2Fnginx%3Apull
  proxy | 2021/07/26 07:53:07 [040] HEAD https://registry.hub.docker.com:443/v2/library/nginx/manifests/1.20.1-alpine
  proxy | 2021/07/26 07:53:07 [040] 200 https://registry.hub.docker.com:443/v2/library/nginx/manifests/1.20.1-alpine
updater | INFO <job_174792754> No update needed for nginx 1.20-alpine
updater | INFO <job_174792754> Finished job processing
updater | time="2021-07-26T07:53:07Z" level=info msg="task complete" container_id=job-174792754-updater exit_code=0 job_id=174792754 step=updater
```

**Comment by deivid-rodriguez** [comment]
Hei! Sorry we didn't get to this before. This will be fixed by #6115!

**Comment by deivid-rodriguez** [comment]
Actually, #6115 won't actually "fix" this.

What you're getting is actually the expected behavior.

Dependabot will never change your "versioning style", so if your Dockerfile specifies `image:<major>.<minor>`, Dependabot will only propose updates that respect that style, namely, updates that bump either the major or the minor version. Since you're explicitly ignore both of those types of update, Dependabot does not propose anything.

If you only want patch updates, you can manually change your Dockerfile to `nginx:1.20.1-alpine`, and then you'll get further patch updates proposed by Dependabot automatically.

I'm closing this as expected behavior but let me know if you have any doubts! Cheers!
